libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -102,21 +97,48 @@ AP_GPS_UBLOX::AP_GPS_UBLOX(AP_GPS &_gps, AP_GPS::GPS_State &_state, AP_HAL::UART
         _unconfigured_messages |= CONFIG_RTK_MOVBASE;
         state.gps_yaw_configured = true;
     }
+#endif
 }
 
 AP_GPS_UBLOX::~AP_GPS_UBLOX()
 {
+#if GPS_UBLOX_MOVING_BASELINE
     if (rtcm3_parser) {
         delete rtcm3_parser;
     }
+#endif
 }
 
+#if GPS_UBLOX_MOVING_BASELINE
 /*
   config for F9 GPS in moving baseline base role
   See ZED-F9P integration manual section 3.1.5.6.1
  */
-const AP_GPS_UBLOX::config_list AP_GPS_UBLOX::config_MB_Base[] {
-#if RTK_MB_UART2 // RTCM3 on UART2
+const AP_GPS_UBLOX::config_list AP_GPS_UBLOX::config_MB_Base_uart1[] {
+ { ConfigKey::CFG_UART2_ENABLED, 0},
```

Comment:
This seems like overstepping, what if a user had already manually set UART2 to feed some other system on the aircraft and saved that config appropriately?

More generally in this case we aren't doing anything with UART2, we shouldn't be configuring/changing any of the settings on that port. It precludes a number of very useful things on the vehicle in this use case.

Reply:
the reason I added this is otherwise if you switch between uart2 and non-uart2 operation and switch rover and base the config will retain the settings, and will leave you pushing half a megabit of rtcm corrections out of a uart you're not using

Reply:
I've changed it to leave uart2 enable along, and only to disable rtcm output on uart2